### PR TITLE
CHE-494: fix websocket machine channels

### PR DESCRIPTION
**Pull request by olexii4**
**_1 Upvote**_ Signed-off-by: Oleksii Orel oorel@codenvy.com

@ashumilova WDYT?

### CODE REVIEW

File: dashboard/src/app/projects/create-project/create-project.controller.js [hunk 1]
```diff
@@ -358,41 +358,30 @@ export class CreateProjectCtrl {
...
-      let agentChannel = 'workspace:' + data.id + ':ext-server:output';
-
-
       let workspaceId = data.id;
 
-      // for now, display log of status channel in case of errors
-      bus.subscribe(statusChannel, (message) => {
-        if (message.eventType === 'DESTROYED' && message.workspaceId === data.id) {
-          this.getCreationSteps()[this.getCurrentProgressStep()].hasError = true;
+      let statusChannel = null;
```
Comment (benoitf) [inline dashboard/src/app/projects/create-project/create-project.controller.js]: if statusChannel is defined later, why do we assign null ?

File: dashboard/src/app/projects/create-project/create-project.controller.js [hunk 2]
```diff
@@ -406,13 +395,45 @@ export class CreateProjectCtrl {
...
+                .content('Unable to start workspace. Error when trying to start the workspace: ' + message.error)
+                .ariaLabel('Workspace start')
+                .ok('OK')
+            );
+          }
+          console.log('Status channel of workspaceID', workspaceId, message);
+        });
+      }
+
+      if(outputChannel){
```
Comment (benoitf) [inline dashboard/src/app/projects/create-project/create-project.controller.js]: missing space after if and before (
missing space between) and {

File: dashboard/src/app/projects/create-project/create-project.controller.js [hunk 3]
```diff
@@ -406,13 +395,45 @@ export class CreateProjectCtrl {
...
       });
 
-      bus.subscribe(outputChannel, (message) => {
-        if (this.getCreationSteps()[this.getCurrentProgressStep()].logs.length > 0) {
-          this.getCreationSteps()[this.getCurrentProgressStep()].logs = this.getCreationSteps()[this.getCurrentProgressStep()].logs + '\n' + message;
-        } else {
-          this.getCreationSteps()[this.getCurrentProgressStep()].logs = message;
-        }
-      });
+      if(statusChannel){
```
Comment (benoitf) [inline dashboard/src/app/projects/create-project/create-project.controller.js]: missing spaces

File: dashboard/src/app/projects/create-project/create-project.controller.js [hunk 4]
```diff
@@ -358,41 +358,30 @@ export class CreateProjectCtrl {
...
 
-          // need to show the error
-          this.$mdDialog.show(
-              this.$mdDialog.alert()
-                  .title('Unable to start workspace')
-                  .content('Unable to start workspace. It may be linked to OutOfMemory or the container has been destroyed')
-                  .ariaLabel('Workspace start')
-                  .ok('OK')
-          );
+        let findStatusLink = this.lodash.find(machineConfigsLinks, function (machineConfigsLink) {
```
Comment (benoitf) [inline dashboard/src/app/projects/create-project/create-project.controller.js]: arrow function is shorter (machineConfigsLink) => {}

File: dashboard/src/app/projects/create-project/create-project.controller.js [hunk 5]
```diff
@@ -358,41 +358,30 @@ export class CreateProjectCtrl {
...
-          // need to show the error
-          this.$mdDialog.show(
-              this.$mdDialog.alert()
-                  .title('Error when starting workspace')
-                  .content('Unable to start workspace. Error when trying to start the workspace: ' + message.error)
-                  .ariaLabel('Workspace start')
-                  .ok('OK')
-          );
+
+        let findOutputLink = this.lodash.find(machineConfigsLinks, function (machineConfigsLink) {
```
Comment (benoitf) [inline dashboard/src/app/projects/create-project/create-project.controller.js]: arrow function

File: dashboard/src/app/projects/create-project/create-project.controller.js [hunk 6]
```diff
@@ -358,41 +358,30 @@ export class CreateProjectCtrl {
...
+          return machineConfigsLink.rel === 'get machine logs channel';
+        });
+        if(findOutputLink) {
```
Comment (benoitf) [inline dashboard/src/app/projects/create-project/create-project.controller.js]: missing space

**Comment by ashumilova** [comment]
+1

**Pull request merged**